src/core/IO/Socket/INET.pm

```diff
@@ -129,15 +135,15 @@ my class IO::Socket::INET does IO::Socket {
 #?endif
         }
         elsif $.type == PIO::SOCK_STREAM {
-            nqp::connect($PIO, nqp::unbox_s($.host), nqp::unbox_i($.port));
+            nqp::connect($PIO, nqp::unbox_s($.host), nqp::unbox_i($.port), nqp::unbox_s($.source-address), nqp::unbox_i($.source-port));
```

Comment:
Particularly here. This can be implemented without changing the `nqp::connect` op.